### PR TITLE
updated testPlan schema and add endpoint

### PR DESCRIPTION
**Pull request by mtran-12**
can send to http://localhost:3000/api/addNewTestPlan with following data to test
{
"name": "LTE Failover",
"detailedDescription": "test description",
"version": "1234",
"deviceConfig": "[{'system':{'system_id':'LTE Router'}}, []]",
"customerFeedback": "Works as expected",
"testCases": ["619003fec01efe48e4b3fe0d"],
"authors": [""],
"isActive": false,
"engagementId": "617246a299915be1b771aca0",
"summaryBOM": [
{
"isOptional": true,
"quantity": 12,
"deviceId": "617251a399915be1b771acb6"
}
]
}

### CODE REVIEW

File: cradlepoint/util/addEntry.js [hunk 1]
```diff
@@ -67,4 +68,28 @@ export async function addTestCase(data) {
     } catch (err) {
         throw err
     }
+}
+
+export async function addTestPlan(data) {
+    try {
+        const client = await connectToDb();
+        const valid = await testPlanSchema.isValid(data);
+        console.log(ObjectId.isValid(data.engagementId))
```
Comment (McGarvs) [inline cradlepoint/util/addEntry.js:77]: Delete the console log

File: cradlepoint/util/addEntry.js [hunk 2]
```diff
@@ -67,4 +68,28 @@ export async function addTestCase(data) {
...
+        if (valid && ObjectId.isValid(data.engagementId)){
+            const testPlan = testPlanSchema.cast(data);
+            const SummaryBOM = testPlan.summaryBOM.map(device => {
+                return {...device, deviceId: ObjectId(device.deviceId)};
+            });
+
+            const result = await client.collection('testPlan').insertOne({...testPlan, SummaryBOM});
+            console.log(result)
```
Comment (McGarvs) [inline cradlepoint/util/addEntry.js:85]: Delete the console log

File: cradlepoint/util/addEntry.js [hunk 3]
```diff
@@ -67,4 +68,28 @@ export async function addTestCase(data) {
...
         throw err
     }
+}
+
+export async function addTestPlan(data) {
+    try {
+        const client = await connectToDb();
+        const valid = await testPlanSchema.isValid(data);
+        console.log(ObjectId.isValid(data.engagementId))
+        if (valid && ObjectId.isValid(data.engagementId)){
```
Comment (McGarvs) [inline cradlepoint/util/addEntry.js:78]: Needs to check that Devices in summaryBOM and that TestCases have valid IDs. I do this in editTestPlan, so you can copy that code.

**Comment by McGarvs** [comment]
The provided json in your comment causes "message": "Input not in right format". Fix this so that I can test

**Comment by mtran-12** [comment]
> The provided json in your comment causes an Error 422 "message": "Input not in right format". Fix this so that I can test

Should be updated to a working version now. name field was missing before

**Pull request merged**